torchvision/csrc/io/decoder/gpu/decoder.cpp

```diff
@@ -138,38 +139,25 @@ int Decoder::handle_picture_display(CUVIDPARSERDISPINFO* disp_info) {
   }
 
   auto options = torch::TensorOptions().dtype(torch::kU8).device(torch::kCUDA);
-  torch::Tensor decoded_frame = torch::empty({get_frame_size()}, options);
+  torch::Tensor decoded_frame =
+      torch::empty({get_height(), get_width(), 3}, options);
```

Comment:
Are you sure we need the rounding by multiples of 2 from https://github.com/pytorch/vision/blob/5e56575e688a85a3bc9dc3c97934dd864b65ce47/torchvision/csrc/io/decoder/gpu/decoder.h#L46 here?

This will give a wrong output shape for the videos which have odd size, so this looks wrong to me. In my prototype I used [width directly](https://github.com/pytorch/vision/pull/5149/files#diff-482fc8a8c571ceb755bd7f447b3803637e06d646a954397f6cf8b0614e2bbe42R143)

Reply:
I think you are right, this should be width as the output format is not NV12.